dp/core/syscall.c

```diff
@@ -199,7 +199,8 @@ static int sys_bpoll(struct bsys_desc __user *d, unsigned int nr)
 				KSTATS_PUSH(idle, NULL);
 
 				start = rdtsc();
-				eth_rx_idle_wait(deadline);
+				if (percpu_get(eth_num_queues) > 0)
```

Comment:
Yes, but this needs to be done in a driver-independent way. A possible way is that the `rx_idle_wait` changes role and only checks and returns the status of the queue.